google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ServerToIDEFileResolver.java

```
@@ -0,0 +1,138 @@
+          .findPackage(getPackageFromPath(path))
+          .findClassByShortName(getClassNameFromPath(path), GlobalSearchScope.allScope(project));
+      if (matchingClasses.length > 0) {
+        file = matchingClasses[0].getContainingFile().getVirtualFile();
+      }
+    }
+    // If we still couldn't find the file, search for possible file name matches and return the
+    // first.
+    // We might want to improve string matching and return more than one possible match.
+    if (file == null) {
+      Collection<VirtualFile> projectJavaFiles = FileBasedIndex.getInstance().getContainingFiles(
+          FileTypeIndex.NAME, JavaFileType.INSTANCE, GlobalSearchScope.projectScope(project));
+      for (VirtualFile projectFile : projectJavaFiles) {
+        if (projectFile.getName().equals(path)) {
+          file = projectFile;
+          break;
+        }
+      }
+    }
+    return file;
+  }
+
+  /**
+   * Produces a Java package name from a file path.
+   *
+   * <p>Example: returns "com.java.package" from "com/java/package/Class.java".
+   */
+  @VisibleForTesting
+  static String getPackageFromPath(String path) {
+    String[] tokens = path.split("/");
+    StringBuilder packageBuilder = new StringBuilder();
+    if (tokens.length > 1) {
+      if(tokens[0].length() > 0) {
+        packageBuilder.append(tokens[0]);
+      }
+      for (int nToken = 1; nToken < tokens.length - 1; nToken++) {
+        if(tokens[nToken].length() > 0 && packageBuilder.length() > 0) {
+          packageBuilder.append(".");
+        }
+        packageBuilder.append(tokens[nToken]);
+      }
+    }
+
+    return packageBuilder.toString();
+  }
+
+  /**
+   * Produces a class name from a file path.
+   *
+   * <p>Example: returns "Class" from "com/java/package/Class.java".
+   */
+  @VisibleForTesting
+  static String getClassNameFromPath(String path) {
+    if (path.indexOf('.') != -1 && path.lastIndexOf("/") < path.lastIndexOf(".")) {
```

if this is false should you throw an exception?